determine_needed_tags.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
+component=$1
+build_version=$2
+registry=registry.hub.docker.com
+tmp_release_cache_file=/tmp/dockerhub_st2_releases
+
+if [[ ${build_version} =~ ^([0-9]+)\.([0-9]+).?([0-9]*)$ ]]; then
+	build_major=${BASH_REMATCH[1]}
+    build_minor=${BASH_REMATCH[2]}
+    build_patch=${BASH_REMATCH[3]}
+else
+	echo "The provided version ${build_version} does not have the expected format."
+	exit 1
+fi
+
+# dockerhub lists the tags in ascending order. 1st object = lowest tag; last object = highest tag
```

Comment:
That's a good point. Until now, I thought that dockerhub will order the tags. I'll run some tests to see what actually happens here. Thanks!

Reply:
@armab I did a bunch of tests now and made the following observations: 

- the returned tags are always ordered by the tag itself NOT by the upload or notification date 
- the first element of the returned list is always the latest tag 

Do we want to rely on it? There seems to be no efficient way in jq to sort the tags because they are strings and not integers. We could do something like piping the jq output to sort the list by ourselves. Any experience with the Docker API and how stable the interface is? Would you expect changes?

Reply:
Yes, I think writing the script without order assumptions would put us on a safe side, as Docker Hub API doesn't advertize any order for the returned data set: https://docs.docker.com/registry/spec/api/#tags

Some Docker Registries implement `orderby` query param like https://docs.microsoft.com/en-us/rest/api/containerregistry/manifests/getlist#uri-parameters

Reply:
Done. We don't rely on the API to provide the tags as an ordered list any longer.